### PR TITLE
[WIP] GH-148: App Actions

### PR DESCRIPTION
**Pull request by edsnider**
### Description of Change ###

This PR adds App Actions API.

### Bugs Fixed ###

- Related to issue #148 

### API Changes ###

List all API changes here (or just put None), example:

Added: 
 
- `ShortcutManager Platform.ShortcutManager { get; }` to `Platform.android.cs`

- New classes:
```csharp
public static partial class AppActions
{
    public static IEnumerable<AppAction> Actions
    {
        get => PlatformGetActions();
        set => PlatformSetActions(value);
    }
}
```
```csharp
public class AppAction
{
    public AppAction(string actionType, string title, string subtitle = null, string icon = null, Uri uri = null)
    {
        ActionType = actionType;
        Title = title;
        Subtitle = subtitle;
        Icon = icon;
        Uri = uri;
    }

    public string ActionType { get; set; }
    public string Title { get; set; }
    public string Subtitle { get; set; }
    public Uri Uri { get; set; }
    internal string Icon { get; set; }
}
```

### Behavioral Changes ###

None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by edsnider** [comment]
Will commit samples and tests next, but wanted to get initial feedback on the implementation.

**Comment by Redth** [comment]
I have a couple of questions / thoughts:

I think we can just name the title properties without the 'Localized' part of the name (e.g. `Title`, and `Subtitle`.

I'm a bit torn on the Icon... I don't love that the property exists on the result when we can't reasonably get the value back when we ask for the existing actions...  I'm almost wondering if the AppAction type should take the icon as a ctor parameter but not expose the value as a readable property.  This would make it very intentional that you can set the Icon when creating the AppAction but that there is no expectation you can read what the icon is when asking for the existing app actions.

**Comment by edsnider** [comment]
I'll remove `Localized` from the title and subtitle properties.

As far as the `Icon` property not being available when "getting".... I like your constructor argument idea. The other thought I had about it was... do we even need to "get" existing actions? Could the API just be the ability to set them?

**Comment by Redth** [comment]
I had the same thought of if we really needed to be able to "Get" them or not, but I think it's a pretty simple bit of code you already have implemented to do so, we just need to deal with the icon.

Another thought around getting actions, will getting them using the API's you've built also return actions declared in the info.plist on iOS and/or the android resource files?  If they do return those, that makes a better case for implementing the 'Get' functionality.

**Comment by edsnider** [comment]
No, the static shortcut items are not returned. On iOS they aren't returned by the API. On Android, they are available via a separate method (`ManifestShortcuts` vs `DynamicShortcuts`) but I'm only returning the dynamic shortcuts for consistency.

**Comment by Redth** [comment]
Ok, I think it would still be nice to have the API return the items even if we have to make the icon available as part of the ctor only.  I think it's a wise choice to not use`ManifestShortcuts` on android if there's no iOS equivalent.

If you can make that change I think this will be ready to go :)

**Comment by Redth** [comment]
@edsnider this looks good, just needs some tests and docs.  Let us know if you need help with those!

**Comment by edsnider** [comment]
Awesome. I've got tests locally... was waiting until we had the model changes locked down before committing those. Will get those committed. Do you want me to update the Sample app as well?

**Comment by Redth** [comment]
If you’re able and willing yes please!

**Comment by edsnider** [comment]
I'm running into build issues related to `ShouldShowRationale()`.

**Comment by edsnider** [comment]
@Redth thanks, I'm able to build again... I'll get the docs and sample committed as soon as possible.

**Comment by MartinZikmund** [comment]
Can you add UWP support too? The `JumpList` API represents this functionality :-) .

File: Xamarin.Essentials/AppActions/AppActions.uwp.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            if (!string.IsNullOrEmpty(action.Subtitle))
+                item.Description = action.Subtitle;
+
+            if (!string.IsNullOrEmpty(action.Icon))
+            {
+                var dir = IconDirectory.Trim('/', '\\').Replace('\\', '/');
+
+                var ext = IconExtension.TrimStart('.');
+
+                item.Logo = new Uri($"ms-appx:///{dir}/{action.Icon}.{ext}");
```
Comment (mattleibow) [inline Xamarin.Essentials/AppActions/AppActions.uwp.cs:95]:
```suggestion
                var ext = IconExtension;
                if (!ext.StartsWith("."))
                    ext = "." + ext;

                item.Logo = new Uri($"ms-appx:///{dir}/{action.Icon}{ext}");
```

**Pull request merged**